### PR TITLE
chore: switch to new dpos2.0

### LINKED ISSUE #1743: switch to new DPOS 2.0 contract

### PR DESCRIPTION
**Pull request by MatusKysel**
fixes #1743

### CODE REVIEW

File: tests/pbft_manager_test.cpp [hunk 1]
```diff
@@ -46,24 +46,24 @@ void check_2tPlus1_validVotingPlayers_activePlayers_threshold(size_t committee_s
...
+  //   nodes[0]->getTransactionManager()->insertTransaction(trx);
+  //   trxs_count++;
+  //   EXPECT_HAPPENS({120s, 1s}, [&](auto &ctx) {
+  //     for (auto &node : nodes) {
+  //       if (ctx.fail_if(!node->getFinalChain()->transaction_location(trx->getHash()))) {
+  //         return;
+  //       }
+  //     }
+  //   });
+  // }
```
Comment (gaoqi8649) [inline tests/pbft_manager_test.cpp:66]: The part of code is for delegate different amount of stake to nodes. I don't understand why comment out

File: tests/pbft_manager_test.cpp [hunk 2]
```diff
@@ -391,31 +391,31 @@ TEST_F(PbftManagerTest, check_get_eligible_vote_count) {
...
+
+  //   trxs_count++;
+  //   EXPECT_HAPPENS({120s, 1s}, [&](auto &ctx) {
+  //     for (auto &node : nodes) {
+  //       if (ctx.fail_if(!node->getFinalChain()->transaction_location(trx->getHash()))) {
+  //         return;
+  //       }
+  //     }
+  //   });
+  // }
```
Comment (gaoqi8649) [inline tests/pbft_manager_test.cpp:418]: Same here

**Review by gaoqi8649 — COMMENTED** [review]
I think we should keep hardfork functionality. we may use it later

File: tests/pbft_manager_test.cpp (hunk 1 above)
Comment (MatusKysel) [reply tests/pbft_manager_test.cpp:66]: we do not have same kind of transactions to delegate, it's commented out as this needs to be reworked

**Comment by MatusKysel** [comment]
> I think we should keep hardfork functionality. we may use it later

I did not removed it just disabled for now as it does not have support in new contract

**Review by JakubFornadel — APPROVED** [review]

**Pull request merged**